### PR TITLE
Add common component DashboardNewDeploy for handling refresh of page

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/877

I've also removed the `FeedbackPromptModal` on V1 Project screen as it doesn't appear like it was used anywhere. Can look to refactor further if we want this

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          {handle ?? 'Project'} will be available soon! Try refreshing the page
+          shortly.
+        </Trans>
+        <br />
+        <br />
+        <Button type="primary" onClick={removeModalSearchQueryAndRefesh}>
+          <Trans>Refresh</Trans>
+        </Button>
+      </h2>
+      <FeedbackPromptModal
```
Comment (johnnyd-eth) [inline src/components/shared/Dashboard/DashboardNewDeploy.tsx:70]: Just remove this modal entirely. We haven't been getting any feedback from it, and we're planning to make a generic 'next steps' modal to show after deploy.

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  projectId,
+  owner,
+  initialFeedbackModalOpen = false,
+  isV2 = false,
+}: {
+  handle?: string
+  projectId?: BigNumber
+  owner: string | undefined
+  initialFeedbackModalOpen?: boolean
+  isV2?: boolean
```
Comment (johnnyd-eth) [inline src/components/shared/Dashboard/DashboardNewDeploy.tsx:24]: These last 2 props will be unnecessary once the feedback modal is removed

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import FeedbackPromptModal from 'components/v1/V1Project/modals/FeedbackPromptModal'
+
+import { useState } from 'react'
+
+import { useHistory } from 'react-router-dom'
+
+import { layouts } from 'constants/styles/layouts'
+import { padding } from 'constants/styles/padding'
+
+export default function DashboardNewDeploy({
```
Comment (johnnyd-eth) [inline src/components/shared/Dashboard/DashboardNewDeploy.tsx:13]: **Suggestion:** I don't think this name reflects what this component really does, perhaps something more like 'NewDeployNotAvailable'

Comment (johnnyd-eth) [inline src/components/shared/Dashboard/DashboardNewDeploy.tsx:13]: Suggestion: I don't think this name reflects what this component really does, perhaps something more like 'NewDeployNotAvailable'

**Review by johnnyd-eth — CHANGES_REQUESTED** [review]
Nice one @wraeth-eth all functioning perfectly. Just get rid of everything to do with the feedback modal then we'll get this up. 

Also a note for future: we want to avoid ever passing in something like `version` or `isV2` to shared components. In this case we'd prefer something like `baseProjectURL` or something

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx (hunk 1 above)
Comment (wraeth-eth) [reply src/components/shared/Dashboard/DashboardNewDeploy.tsx]: Removed it

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx (hunk 2 above)
Comment (wraeth-eth) [reply src/components/shared/Dashboard/DashboardNewDeploy.tsx]: Have reworked - now takes in optional `name` and `projectId`. Preference is to use `name` and fallback to `projectId` if a `name` cannot be found.

`name` is used in v1 for `handle` and in v2, will use the metadata name if available, or fallback to projectId

File: src/components/shared/Dashboard/DashboardNewDeploy.tsx (hunk 3 above)
Comment (wraeth-eth) [reply src/components/shared/Dashboard/DashboardNewDeploy.tsx]: Have updated with `NewDeployNotAvailable`

**Comment by wraeth-eth** [comment]
Hey @johnnyd-eth - just had a thought; in `NewDeployNotAvailable`, I am taking in a `name` and `projectId` and making the choice in the component of what to show. I am now thinking that maybe I should just have a `name` argument that is `string | BigNumber` and leave it up to the consumer to pass it in, i.e
```tsx
<NewDeployNotAvailable
  name={projectMetadata?.name ?? projectId}
/>
```

I might just go ahead and make it. If you disagree, can easily revert the commit

File: src/components/v2/V2Dashboard/index.tsx [hunk 4]
```diff
@@ -136,8 +144,11 @@ export default function V2Dashboard({ projectId }: { projectId: BigNumber }) {
   const { data: ballotState } = useBallotState(projectId)
 
   if (metadataLoading || metadataURILoading) return <Loading />
-  if (projectId?.eq(0) || metadataError || !metadataCID) {
-    return <Dashboard404 projectId={projectId} />
+  if (isNewDeploy && !metadataCID) {
+    return <NewDeployNotAvailable name={projectMetadata?.name ?? projectId} />
```
Comment (johnnyd-eth) [inline src/components/v2/V2Dashboard/index.tsx:148]: **Issue:** Good thought here but if `metadataCID` isn't loaded it'll be impossible to get `projectMetadata?.name` anyway so I think should just pass `projectId` only

File: src/components/v2/V2Dashboard/index.tsx [hunk 5]
```diff
@@ -136,8 +144,11 @@ export default function V2Dashboard({ projectId }: { projectId: BigNumber }) {
...
+  }
+  if (metadataError || !metadataCID) {
+    return <Project404 projectId={projectId} />
```
Comment (johnnyd-eth) [inline src/components/v2/V2Dashboard/index.tsx:151]: **Praise**: this is great, nice one

File: src/components/shared/NewDeployNotAvailable.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { layouts } from 'constants/styles/layouts'
+import { padding } from 'constants/styles/padding'
+
+export default function NewDeployNotAvailable({
+  name,
+}: {
+  name?: string | BigNumber
```
Comment (johnnyd-eth) [inline src/components/shared/NewDeployNotAvailable.tsx:14]: Yeah this is good, but maybe now that we're not passing `projectMetadata.name` from `V2Dashboard`,  we can change this `name` prop to `handleOrId` ?

Comment (wraeth-eth) [reply src/components/shared/NewDeployNotAvailable.tsx]: Done 💥

File: src/components/v2/V2Dashboard/index.tsx (hunk 4 above)
Comment (wraeth-eth) [reply src/components/v2/V2Dashboard/index.tsx]: Done 🌔

**Review by johnnyd-eth — APPROVED** [review]

**Comment by johnnyd-eth** [comment]
Nice one @wraeth-eth merging this

**Pull request merged**